data-structures-queues/take-next-smallest.js

```diff
@@ -0,0 +1,18 @@
+/* exported takeNextSmallest */
+
+function takeNextSmallest(queue) {
+  if (typeof queue.peek() === 'undefined') return;
+
+  let current = queue.dequeue();
+
+  while (true) {
+    if (typeof queue.peek() !== 'undefined') {
```

Comment:
Don't do this, passing `true` to a `while` loop is never a good idea. I know people do it but there is always a better solution.